scripts/pre_push_hook.sh

```diff
@@ -0,0 +1,125 @@
+
+# Hook file paths, we need you use relative path as actual path won't be same in host and container
+PYTHON_PREPUSH_HOOK_PATH=".git/hooks/pre-push-python"
+PREPUSH_HOOK_PATH=".git/hooks/pre-push"
+
+# Get the oppia app directory
+OPPIA_DIR=$(pwd)
+
+# Location of git hooks directory
+HOOKS_DIR="$OPPIA_DIR/.git/hooks"
+
+# Path for pre-push hook file
+PRE_PUSH_FILE="$HOOKS_DIR/pre-push"
+PYTHON_PRE_PUSH_FILE="$HOOKS_DIR/pre-push-python"
+
+# Install pre-push hook
+install_hook() {
+    # Create symlinks for pre-push hook files
+    for file in "$PRE_PUSH_FILE" "$PYTHON_PRE_PUSH_FILE"; do
+        # Check if pre-push file is already a symlink
+        if [ -h "$file" ] && [ -e "$file" ]; then
+            echo "Symlink already exists (for $file)"
+
+        else
+            # If broken symlink, remove it
+            if [ -h "$file" ] && [ ! -e "$file" ]; then
+                rm "$file"
+                echo "Removed broken symlink (for $file)"
+            fi
+
+            # Try creating a symlink
+            base_symlink="../../scripts"  # Symlinks needs to be relative as actual path won't be same on host and docker container
+            if [ "$(basename $file)" == "pre-push" ]; then
+                ORIGINAL_FILE="$base_symlink/pre_push_hook.sh"  
+            else
+                ORIGINAL_FILE="$base_symlink/pre_push_hook.py"
+            fi
+
+                     ln -s "$ORIGINAL_FILE" "$file" &&
+                echo "Created symlink in .git/hooks directory" ||
+                {
+                    # Fallback to copy on windows
+                    cp "$ORIGINAL_FILE" "$file"
+                    echo "Copied file to .git/hooks directory"
+                }
+
+            # Make the hook file executable
+            chmod +x "$file"
+
+            if [ $? -eq 0 ]; then
+                echo "pre-push hook file is now executable!"
+            else
+                echo >&2 "Failed to make pre-push executable"
+                exit 1
+            fi
+        fi
+    done
+    exit 0
+}
+
+# Check for --install in args and install pre-push hook if itC's found
+for arg in "$@"; do
+    if [ "$arg" == "--install" ]; then
+        install_hook
+    fi
+done
+
+# Bypassing notification
+echo "You can Bypass Pre-Push check by using \`git push --no-verify\`"
+
+# Check if dev-server is running and is healthy
+$(docker ps -a --format '{{json .}}' | grep $DEV_CONTAINER | jq .Status | grep -q healthy)
+is_dev_server_running=$?
+
+if [ "$is_dev_server_running" != "0" ]; then
+    # Start containers and run pre-push hook
+    make start-devserver  # We don't need to use run-offline as internet would be available when pushing commit
+fi
+
+# Run hook in container
+CMD="$DOCKER_EXEC_COMMAND python3 ./$PYTHON_PREPUSH_HOOK_PATH $@"
+echo "Running $CMD"
+
+$CMD
+
+# Save exit code from the docker command
+exitcode=$?
+echo "Python script exited with code $exitcode"
+
+# Shut down containers if they were not running before
```

Comment:
Why stop the containers if they're not running?

Reply:
I meant if containers were running before running hook. Updated variable name to `was_container_running_before_hook` and changed comment to `# Shut down containers if they were not running before running hook`

Reply:
done?